### PR TITLE
Hint Changes

### PR DESCRIPTION
**Pull request by johnchristianbanzon**

### CODE REVIEW

File: Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartAnswerScripts/FillAnswerType.cs [hunk 1]
```diff
@@ -2,39 +2,54 @@
...
 	private string questionAnswer;
 	private int answerIndex = 0;
 	Action<bool> onHintResult;
+	List <int> hintIndexRandomList = new List<int> ();
+	public string answerWrote = "";
+
 	public void DeployAnswerType(){
 		gameObject.SetActive (true);
+		hintIndexRandomList.Clear ();
 		this.questionAnswer = QuestionSystemController.Instance.questionAnswer;
```
Comment (jktarban) [inline Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartAnswerScripts/FillAnswerType.cs]: this?

File: Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartAnswerScripts/FillAnswerType.cs [hunk 2]
```diff
@@ -2,39 +2,54 @@
...
+
 public class FillAnswerType : MonoBehaviour,IAnswer {
 	
 	public List<GameObject> answerContainers = new List<GameObject>();
-	private GameObject[] selectionIdentifier = new GameObject[12];
+	public GameObject[] selectionIdentifier = new GameObject[12];
 	public GameObject outviewContent;
 	private string questionAnswer;
 	private int answerIndex = 0;
 	Action<bool> onHintResult;
```
Comment (jktarban) [inline Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartAnswerScripts/FillAnswerType.cs]: put private/public in all scripts

File: Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartAnswerScripts/FillAnswerType.cs [hunk 3]
```diff
@@ -76,23 +89,29 @@ public void OnAnswerClick (Button answerButton)
...
+			hintIndexRandomList.Add (answerIndex);
 			answerContainers [answerIndex].GetComponentInChildren<Text>().text 
 			= selectedObject.GetComponentInChildren<Text>().text;
 			CheckAnswer ();
 		}
 	}
 
-	public void CheckAnswer(){
-		string answerWrote = "";
+	public void GetAnswerWritten(){
```
Comment (jktarban) [inline Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartAnswerScripts/FillAnswerType.cs]: only use Get Prefix if there is return value... apply to all scripts also

File: Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartSelectionScripts/LetterLink.cs [hunk 4]
```diff
@@ -14,30 +15,34 @@ public class LetterLink : MonoBehaviour ,ISelection{
...
 	private int noOfContainersSelected = 0;
 
-	public void ShowCorrectAnswer(){
+	public void ShowCorrectAnswer ()
+	{
 		
 	}
 
-	public void OnbeginDrag(GameObject currentSelectedLetter){
+	public void OnbeginDrag (GameObject currentSelectedLetter)
```
Comment (jktarban) [inline Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartSelectionScripts/LetterLink.cs]: OnBeginDrag

File: Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartSelectionScripts/SelectLetter.cs [hunk 5]
```diff
@@ -3,26 +3,40 @@
...
-		QuestionSystemController.Instance.partAnswer.fillAnswer.SelectionLetterGot(EventSystem.current.currentSelectedGameObject);
-		EventSystem.current.currentSelectedGameObject.GetComponentInChildren<Text> ().text = "";
-//		EventSystem.current.currentSelectedGameObject.SetActive (false);
+	public void OnSelect ()
+	{
+		QuestionSystemController.Instance.partAnswer.fillAnswer.GetAnswerWritten ();
+		if (questionAnswer.Length > QuestionSystemController.Instance.partAnswer.fillAnswer.answerWrote.Length) {
+			QuestionSystemController.Instance.partAnswer.fillAnswer.SelectionLetterGot (EventSystem.current.currentSelectedGameObject);
+			EventSystem.current.currentSelectedGameObject.SetActive (false);
+		} else {
```
Comment (jktarban) [inline Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartSelectionScripts/SelectLetter.cs]: empty else

File: Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartSelectionScripts/SelectLetter.cs [hunk 6]
```diff
@@ -31,11 +45,29 @@ public class SelectLetter : MonoBehaviour, ISelection
...
+		GameObject letterHint = null;
+		GameObject[] selectionArray = QuestionSystemController.Instance.partSelection.selectLetter.selectionButtons;
+		for (int i = 0; i < selectionArray.Length; i++) {
+			if (selectionArray [i].GetComponentInChildren<Text> ().text.Equals (questionAnswer [hintIndex].ToString ())) {
+				letterHint = selectionArray [i];
+				break;
+			}
+		}
+
+		if (!correctAnswerContainer.GetComponentInChildren<Text> ().text.Equals ("")) {
```
Comment (jktarban) [inline Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartSelectionScripts/SelectLetter.cs]: use string.isnullorempty

File: Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartAnswerScripts/FillAnswerType.cs (hunk 1 above)
Comment (johnchristianbanzon) [reply Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartAnswerScripts/FillAnswerType.cs]: done

File: Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartAnswerScripts/FillAnswerType.cs (hunk 2 above)
Comment (johnchristianbanzon) [reply Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartAnswerScripts/FillAnswerType.cs]: done

File: Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartSelectionScripts/SelectLetter.cs (hunk 5 above)
Comment (johnchristianbanzon) [reply Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartSelectionScripts/SelectLetter.cs]: done

File: Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartSelectionScripts/LetterLink.cs (hunk 4 above)
Comment (johnchristianbanzon) [reply Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartSelectionScripts/LetterLink.cs]: Done

File: Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartAnswerScripts/FillAnswerType.cs (hunk 3 above)
Comment (johnchristianbanzon) [reply Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartAnswerScripts/FillAnswerType.cs]: Done

File: Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartSelectionScripts/SelectLetter.cs (hunk 6 above)
Comment (johnchristianbanzon) [reply Assets/EnglishBattleRoyale/Scripts/QuestionSystem/PartSelectionScripts/SelectLetter.cs]: Done

**Pull request closed without merging**